btr-web/btr-common-components/utils/canadaPostAddressApi/find-v2.10.ts

```diff
@@ -0,0 +1,63 @@
+const CANADA_POST_FIND_API_URL =
+  'https://ws1.postescanada-canadapost.ca/AddressComplete/Interactive/Find/v2.10/json3.ws'
+
+// https://www.canadapost-postescanada.ca/ac/support/api/addresscomplete-interactive-find/#parameters
+// API : Find (v2.10)
+export type CanadaPostApiFindParams = {
+  // mandatory, error if not provided
+  Key: string
+  SearchTerm: string
+
+  // with defaults if not provided
+  LastId?: string // default ''
+  Country?: string // default 'CAN'
+  LanguagePreference?: string // default 'en'
+  MaxSuggestions?: Number // default 7
+
+  // in examples, but could not find description of it
+  SearchFor?: string
+  MaxResults?: Number
+  Origin?: string
+  Bias?: string
+  Filters?: string
+  GeoFence?: string
+}
+
+// https://www.canadapost-postescanada.ca/ac/support/api/addresscomplete-interactive-find/#responses
+// API : Find (v2.10)
+export type CanadaPostApiFindResponseItem = {
+  Id: string // use as LastId in find method or as id to select the item with retrieve
+  Text: string // suggestion text
+  Highlight: string
+  Cursor: Number
+  Description: string
+  Next: 'Find' | 'Retrieve'
+}
+
+export const findAddress = async (searchTerm: string,
+  lastId: string,
+  props: { countryIso3166Alpha2: string, maxSuggestions: Number, langCode: string },
+  canadaPostApiKey: string) => {
+  if (!searchTerm) {
+    return
+  }
+  const searchParams: CanadaPostApiFindParams = {
+    Key: canadaPostApiKey,
+    SearchTerm: searchTerm,
+    LastId: lastId || '',
+    MaxSuggestions: props.maxSuggestions,
+    LanguagePreference: props.langCode,
+    Country: props.countryIso3166Alpha2
+  }
+
+  // @ts-ignore
```

Comment:
can you comment why we are doing the ts ignore here and below or ignore the specific ts rule?

Reply:
Both are about types used.

For this one searchParams do not "match" URLSearchParams, which is effectively any object with string values. And searchParams have one declared as Number which gets conveted to string in new URLSearchParams... Javascript magic ...

For bellow one didnt want to stick type to the await command. It would be something like
<type>await $fetch ...